packages/react-instantsearch-core/src/core/createConnector.tsx

```diff
@@ -0,0 +1,385 @@
+        ['getMetadata', 'getSearchParameters', 'refine', 'cleanUp'].indexOf(
+          key
+        ) > -1
+    );
+
+    if (
+      onlyGetProvidedPropsUsage &&
+      connectorDesc.displayName.substr(0, 7) !== 'Algolia'
+    ) {
+      // tslint:disable-next-line no-console
+      console.warn(
+        'react-instantsearch: it seems that you are using the `createConnector` api ' +
+          'only to access the `searchState` and the `searchResults` through `getProvidedProps`.' +
+          'We are now provided a dedicated API' +
+          ' the `connectStateResults` connector that you should use instead. The `createConnector` API will be ' +
+          'soon deprecated and will break in future next major versions.' +
+          '\n\n' +
+          'See more at https://www.algolia.com/doc/api-reference/widgets/state-results/react/' +
+          '\n' +
+          'and https://www.algolia.com/doc/guides/building-search-ui/going-further/conditional-display/react/'
+      );
+    }
+  }
+}
+
+export type ConnectorDescription = {
+  displayName: string;
+  /**
+   * a function to filter the local state
+   */
+  refine?: (...args: any[]) => any;
+  /**
+   * function transforming the local state to a SearchParameters
+   */
+  getSearchParameters?: (...args: any[]) => any;
+  /**
+   * metadata of the widget (for current refinements)
+   */
+  getMetadata?: (...args: any[]) => any;
+  /**
+   * hook after the state has changed
+   */
+  transitionState?: (...args: any[]) => any;
+  /**
+   * transform the state into props passed to the wrapped component.
+   * Receives (props, widgetStates, searchState, metadata) and returns the local state.
+   */
+  getProvidedProps: (...args: any[]) => any;
+  /**
+   * Receives props and return the id that will be used to identify the widget
+   */
+  getId?: (...args: any[]) => string;
+  /**
+   * hook when the widget will unmount. Receives (props, searchState) and return a cleaned state.
+   */
+  cleanUp?: (...args: any[]) => any;
+  searchForFacetValues?: (...args: any[]) => any;
+  shouldComponentUpdate?: (...args: any[]) => boolean;
+  /**
+   * PropTypes forwarded to the wrapped component.
+   */
+  propTypes?: {}; // I can't find a definition for a propTypes object
+  defaultProps?: {};
+};
+
+/**
+ * Connectors are the HOC used to transform React components
+ * into InstantSearch widgets.
+ * In order to simplify the construction of such connectors
+ * `createConnector` takes a description and transform it into
+ * a connector.
+ * @param {ConnectorDescription} connectorDesc the description of the connector
+ * @return {Connector} a function that wraps a component into
+ * an instantsearch connected one.
+ */
+export function createConnectorWithoutContext(
+  connectorDesc: ConnectorDescription
+) {
+  if (!connectorDesc.displayName) {
+    throw new Error(
+      '`createConnector` requires you to provide a `displayName` property.'
+    );
+  }
+
+  needlessUsageWarning(connectorDesc);
```

Comment:
We can remove the message, it's here since a long time and we advocate for the usage of `stateResults` a lot inside the doc. It should not be an issue anymore.